draft-ietf-tls-ctls.md

```diff
@@ -237,36 +237,54 @@ defined in {{RFC8446, Section 8.4}}.
 
 #### `dh_group`
 
-Value: a single `NamedGroup` ({{!RFC8446, Section 4.2.7}}) to use for key establishment.
+Value: a single `CTLSKeyShareGroup` to use for key establishment.
 
-This is equivalent to a literal "supported_groups" extension
-consisting solely of this group.
+~~~~
+struct {
+    NamedGroup group_name;
+    uint16 key_share_length;
+} CTLSKeyShareGroup;
+~~~~
+
+This is equivalent to adding a "supported_groups" extension to every message
+where that is allowed (i.e. ClientHello and EncryptedExtensions, in TLS 1.3)
+consisting solely of the group `CTLSKeyShareGroup.group_name`.
 
 Static vectors (see {{static-vectors}}):
 
 * `KeyShareClientHello.client_shares`
-* `KeyShareEntry.key_exchange`, if the `NamedGroup` uses fixed-size key shares.
+* `KeyShareEntry.key_exchange`, if `CTLSKeyShareGroup.key_share_length` is non-zero.
 
-In JSON, the group is listed by the code point name in {{RFC8446, Section 4.2.7}}
-(e.g., "x25519").
+In JSON, this value is represented as a dictionary with two keys:
+* `groupName`: a string containing the code point name from the TLS Supported Groups registry (e.g., "x25519").
+* `keyShareLength`: an integer, defaulting to zero if omitted.
```

Comment:
I think I must be missing something here. Can you explain verbally how this is intended to work?

Reply:
This sentence should be added to the draft.

Reply:
We can add it as a reminder, but as a formal matter it is already covered by line 256 and the definition of a "static vector".